packages/core/src/rum/instrumentation/resourceTracking/distributedTracing/distributedTracing.tsx

```diff
@@ -77,55 +109,177 @@ const generateTracingAttributesWithSampling = (
  * Using branded types will ensure we don't accidentally use
  * traceId for spanId when generating headers.
  */
-export type TraceId = TraceIdentifier & {
+export type TraceId = TracingIdentifier & {
     _brand: 'traceId';
 };
 
-export type SpanId = TraceIdentifier & {
+export type SpanId = TracingIdentifier & {
     _brand: 'spanId';
 };
 
-/*
- * This code was inspired from browser-sdk at (https://github.com/DataDog/browser-sdk/blob/0e9722d5b06f6d49264bc82cd254a207d647d66c/packages/rum-core/src/domain/tracing/tracer.ts#L190)
- */
-const MAX_32_BITS_NUMBER = 4294967295; // 2^32-1
-const MAX_31_BITS_NUMBER = 2147483647; // 2^31-1
-export class TraceIdentifier {
-    private low: number;
-    private high: number;
+export class TracingIdentifier {
+    private id: string;
+    private high: string;
+    private low: string;
 
     constructor() {
-        // We need to have a 63 bits number max
-        this.high = Math.floor(Math.random() * MAX_31_BITS_NUMBER);
-        this.low = Math.floor(Math.random() * MAX_32_BITS_NUMBER);
+        const idData = this.generate128BitID();
+        this.id = idData.id;
+        this.high = idData.high;
+        this.low = idData.low;
     }
 
-    toString = (radix: number) => {
-        let low = this.low;
-        let high = this.high;
-        let str = '';
+    /**
+     * Generates a 128-bit ID in the format:
+     * <32-bit Unix timestamp> <32 bits of zero> <64 random bits>.
+     * @returns An object containing the full 128-bit ID, high 64 bits, and low 64 bits as hexadecimal strings.
+     */
+    private generate128BitID(): { id: string; high: string; low: string } {
```

Comment:
Could we possibly generate the ID as a BigInt, which would save re-parsing it when writing out the representations.